### PR TITLE
Change password

### PR DESCRIPTION
**Pull request by piobra**
@katmatt please, review
#395

### CODE REVIEW

File: my-account/app/com/commercetools/sunrise/myaccount/authentication/changepassword/viemodels/ChangePasswordPageContentFactory.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void initialize(final ChangePasswordPageContent viewModel, final Void input, final Form<? extends ChangePasswordFormData> form) {
+        super.initialize(viewModel, input, form);
+        fillChangePasswordForm(viewModel, form);
+    }
+
+    @Override
+    protected void fillTitle(final ChangePasswordPageContent viewModel, final Void input, final Form<? extends ChangePasswordFormData> form) {
+        viewModel.setTitle(pageTitleResolver.getOrEmpty("my-account:changePassword.title")); // TODO messages missing in FE templates
```
Comment (katmatt) [inline my-account/app/com/commercetools/sunrise/myaccount/authentication/changepassword/viemodels/ChangePasswordPageContentFactory.java]: We have a translation for the title in our [my-account.yaml](https://github.com/commercetools/commercetools-sunrise-theme/blob/master/input/i18n/en/my-account.yaml#L86), but it's called `changePasswordTitle`. We should later rename it to `title`, but for now I would suggest to use `my-account:changePassword.changePasswordTitle` here.

File: common/app/com/commercetools/sunrise/framework/reverserouters/myaccount/authentication/AuthenticationLinksControllerComponent.java [hunk 2]
```diff
@@ -26,5 +26,7 @@ protected void addLinksToPage(final PageMeta meta, final AuthenticationReverseRo
         meta.addHalLink(reverseRouter.logOutProcessCall(), "logOut");
         meta.addHalLink(reverseRouter.logInPageCall(), "signUp");
         meta.addHalLink(reverseRouter.signUpProcessCall(), "signUpSubmit");
+        meta.addHalLink(reverseRouter.changePasswordPageCall(), "changePassword");
```
Comment (katmatt) [inline common/app/com/commercetools/sunrise/framework/reverserouters/myaccount/authentication/AuthenticationLinksControllerComponent.java]: Additionally I would suggest to introduce a separate `ChangePasswordLinksControllerComponent` class and a corresponding `ChangePasswordReverseRouter` interface with a `DefaultChangePasswordReverseRouter` class.

**Comment by piobra** [comment]
@katmatt thanks for remarks.
I made adjustments according to them.
Regarding routes I changed to "changepassword" rather than "password" because there will be also reset password routes. I'm currently working on them. I'm thinking about making them "resetpassword".

If I did it your way we'd end up with sth like "password" and "reset" which in my opinion is not clear and self descriptive.

**Comment by katmatt** [comment]
@piobra I see why you choose to use `changepassword`, but I still would suggest to use `password`and later to add `password/reset` for reseting the password.

With regards to working on the reset/recover password feature: I think that this feature requires more preparation (e.g. sending a confirmation link etc.) and I don't think that it makes sense to work on it now. And that's what we already discussed with Florian.

**Comment by piobra** [comment]
@katmatt done

**Pull request merged**